x-pack/plugins/alerting/server/alerts_client/lib/build_new_alert.ts

```diff
@@ -36,30 +39,41 @@ export const buildNewAlert = <
   legacyAlert,
   rule,
   timestamp,
-}: BuildNewAlertOpts<LegacyState, LegacyContext, ActionGroupIds, RecoveryActionGroupId>): Alert &
-  AlertData => {
-  return {
-    '@timestamp': timestamp,
-    kibana: {
-      alert: {
-        action_group: legacyAlert.getScheduledActionOptions()?.actionGroup,
-        flapping: legacyAlert.getFlapping(),
-        instance: {
-          id: legacyAlert.getId(),
+  payload,
+}: BuildNewAlertOpts<
+  AlertData,
+  LegacyState,
+  LegacyContext,
+  ActionGroupIds,
+  RecoveryActionGroupId
+>): Alert & AlertData => {
+  const cleanedPayload = payload ? stripFrameworkFields(payload) : {};
+  return deepmerge.all(
```

Comment:
I tested this out, generating 1000 alerts and it seemed to take between 60 to ~200 ms. Not great but doesn't seem terrible to me either. I'm not sure what our alternatives are since we need a deep object merge without merging array entries, which lodash `merge` and `defaultsDeep` does not provide.